src/io/DumpH5MDParallel.cpp

```diff
@@ -0,0 +1,479 @@
+                                   count.data(), localDims.data()));
+
+    auto datawrite = CHECK_HDF5(H5Pcreate(H5P_DATASET_XFER));
+    CHECK_HDF5(H5Pset_dxpl_mpio(datawrite, H5FD_MPIO_COLLECTIVE));
+    CHECK_HDF5(H5Dwrite(dataset, typeToHDF5<T>(), srcSpace, dstSpace, datawrite, data.data()));
+
+    CHECK_HDF5(H5Pclose(datawrite));
+    CHECK_HDF5(H5Sclose(dstSpace));
+    CHECK_HDF5(H5Sclose(srcSpace));
+    CHECK_HDF5(H5Dclose(dataset));
+    CHECK_HDF5(H5Sclose(dataspace));
+}
+
+void DumpH5MDParallel::writeHeader(hid_t fileId) const
+{
+    auto group1 = CHECK_HDF5(H5Gcreate(fileId, "/h5md", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    auto group2 =
+        CHECK_HDF5(H5Gcreate(fileId, "/h5md/author", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    auto group3 =
+        CHECK_HDF5(H5Gcreate(fileId, "/h5md/creator", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    CHECK_HDF5(H5Gclose(group1));
+    CHECK_HDF5(H5Gclose(group2));
+    CHECK_HDF5(H5Gclose(group3));
+
+    std::vector<int> data = {2, 0};
+    CHECK_HDF5(H5LTset_attribute_int(fileId, "/h5md", "version", data.data(), data.size()));
+
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/author", "name", author.c_str()));
+
+    std::string software = "espressopp";
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "name", software.c_str()));
+
+    std::string version = Version().info();
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "version", version.c_str()));
+}
+
+void DumpH5MDParallel::writeBox(hid_t fileId)
+{
+    std::string groupName = "/particles/atoms/box";
+    auto group =
+        CHECK_HDF5(H5Gcreate(fileId, groupName.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    std::vector<int> dims = {3};
+    CHECK_HDF5(
+        H5LTset_attribute_int(fileId, groupName.c_str(), "dimension", dims.data(), dims.size()));
+
+    auto boundaryType = H5Tcopy(H5T_C_S1);
+    CHECK_HDF5(H5Tset_size(boundaryType, 8));
+    CHECK_HDF5(H5Tset_strpad(boundaryType, H5T_STR_NULLPAD));
+    std::vector<hsize_t> boundaryDims = {3};
+    auto space = H5Screate_simple(int_c(boundaryDims.size()), boundaryDims.data(), nullptr);
+    auto att = H5Acreate(group, "boundary", boundaryType, space, H5P_DEFAULT, H5P_DEFAULT);
+    CHECK_HDF5(H5Awrite(att, boundaryType, "periodicperiodicperiodic"));
+    CHECK_HDF5(H5Aclose(att));
+    CHECK_HDF5(H5Sclose(space));
+    CHECK_HDF5(H5Tclose(boundaryType));
+
+    std::vector<double> edges = {system_->bc->getBoxL()[0], system_->bc->getBoxL()[1],
+                                 system_->bc->getBoxL()[2]};
+    CHECK_HDF5(
+        H5LTset_attribute_double(fileId, groupName.c_str(), "edges", edges.data(), edges.size()));
+
+    CHECK_HDF5(H5Gclose(group));
+}
+
+void DumpH5MDParallel::writeId(hid_t fileId)
+{
+    using Datatype = int64_t;
+    constexpr int64_t dimensions = 1;  ///< dimensions of the property
+
+    std::string groupName = "/particles/atoms/" + idDataset;
```

Comment:
Let's keep string `/particles/atoms/` as a variable - for safety

Reply:
introduced a new variable for 'atoms'